protobuf-codegen/src/code_writer.rs

```diff
@@ -270,6 +270,38 @@ impl<'a> CodeWriter<'a> {
         }
     }
 
+    fn documentation(&mut self, comment: &str) {
+        if comment.is_empty() {
+            self.write_line("///");
+        } else {
+            self.write_line(&format!("/// {}", comment));
+        }
+    }
+
+    pub fn all_documentation(
+        &mut self,
+        info: Option<&protobuf::descriptor::SourceCodeInfo>,
+        path: &[i32],
+    ) {
+        let doc = info
+            .map(|v| &v.location)
+            .and_then(|ls| ls.iter().find(|l| l.path == path))
+            .map(|l| l.get_leading_comments());
+
+        let lines = doc
+            .iter()
+            .map(|doc| doc.lines())
+            .flatten()
+            .collect::<Vec<_>>();
+
+        // Skip comments with code blocks to avoid rustdoc trying to compile them.
+        if !lines.iter().any(|line| line.starts_with("    ")) {
```

Comment:
Can't we do something like

```
documentation("```ignore");
...
documentation("```");
```

?

Reply:
... or at least TODO

Reply:
I tried, but then you hit things like

https://github.com/stepancheg/rust-protobuf/blob/0674fbba3a091667f9b345f5a785e2377adab8c4/proto/google/protobuf/descriptor.proto#L385-L391

which do contain 4 space indented lines, but it's not actually the entire block, so how would we know where to add the ` ``` `?. Google's proto files are not very consistent with the number of spaces within the documentation. My attempts at making this work started to get quite complex and seemed out of scope of this PR.